bundles/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastStatusUpdater.java

```diff
@@ -245,21 +246,39 @@ private void updateImage(Map<String, Object> metadata) {
             }
         }
 
-        // Poor man's cache. If the imageSrc is the same, don't update them.
-        if (Objects.equals(this.imageSrc, imageSrc)) {
-            return;
-        } else {
-            this.imageSrc = imageSrc;
-        }
-
         if (callback.isLinked(CHANNEL_IMAGE_SRC)) {
-            callback.updateState(CHANNEL_IMAGE_SRC, imageSrc == null ? UNDEF : new StringType(imageSrc));
+            callback.updateState(CHANNEL_IMAGE_SRC, imageSrc == null ? UnDefType.UNDEF : new StringType(imageSrc));
         }
 
         if (callback.isLinked(CHANNEL_IMAGE)) {
-            State image = imageSrc == null ? UNDEF : HttpUtil.downloadImage(imageSrc);
-            callback.updateState(CHANNEL_IMAGE, image == null ? UNDEF : image);
+            State image = imageSrc == null ? UnDefType.UNDEF : downloadImageFromCache(imageSrc);
+            callback.updateState(CHANNEL_IMAGE, image == null ? UnDefType.UNDEF : image);
+        }
+    }
+
+    private @Nullable RawType downloadImage(String url) {
+        logger.debug("Trying to download the content of URL '{}'", url);
+        RawType downloadedImage = HttpUtil.downloadImage(url);
+        if (downloadedImage == null) {
+            logger.debug("Failed to download the content of URL '{}'", url);
+        }
+        return downloadedImage;
+    }
+
+    private @Nullable RawType downloadImageFromCache(String url) {
+        if (IMAGE_CACHE.containsKey(url)) {
+            byte[] bytes = IMAGE_CACHE.get(url);
+            String contentType = HttpUtil.guessContentTypeFromData(bytes);
+            return new RawType(bytes,
+                    contentType == null || contentType.isEmpty() ? RawType.DEFAULT_MIME_TYPE : contentType);
+        } else {
+            RawType image = downloadImage(url);
+            if (image != null) {
+                IMAGE_CACHE.put(url, image.getBytes());
+                return image;
+            }
```

Comment:
No, I do not think so.

Reply:
Should it be thread-safe?

Reply:
No, I do not think so :wink:. I am using this implementation in four different bindings and never faced an issue related to the cached data. In general it has to handle one channel (approximately one image every three to four minutes).

Reply:
Maybe we should consider adding this to the into the 3.0 core if it used in so many places...

Reply:
I will add it to the list in #4736. I tried it once but it did not work out nicely. My current implementation uses `ConfigConstants.getUserDataFolder()` which is not available in package `org.openhab.core.cache`. I have to solve one of the todos first. There are even bigger plans for it (see e.g. https://github.com/eclipse/smarthome/issues/6041) to make it configurable and it could be used for voice add-ons to cache similar speech data.

Reply:
Sounds good to me